### PR TITLE
Improve Group documentation

### PR DESCRIPTION
**Pull request by giarc3**
First attempt at improving all docs relating to groups. Please give as much feedback as you can.

### CODE REVIEW

File: src/group.rs [hunk 1]
```diff
@@ -141,143 +147,287 @@ impl GroupCreateOpts {
...
 
-    /// Rotate the provided group's private key, but leave the public key the same.
-    /// There's no black magic here! This is accomplished via multi-party computation with the
-    /// IronCore webservice.
-    /// Note: You must be an admin of the group in order to rotate its private key.
+    /// Deletes a group.
+    ///
+    /// A group can be deleted even if it has existing members and administrators.
+    ///
+    /// **Warning: Deleting a group will cause all documents encrypted to that
```
[the remark this answers is not in the document]
Comment (giarc3) [reply src/group.rs:409]: I went with 
`Warning: Deleting a group will prevent its members from decrypting all of the documents previously encrypted to the group. Caution should be used when deleting groups.`
Let me know if you have further notes.

File: src/group.rs [hunk 2]
```diff
@@ -141,143 +147,287 @@ impl GroupCreateOpts {
 }
 
 impl Default for GroupCreateOpts {
-    /// Default GroupCreateOpts for common use cases. The user who calls `group_create()` will be the owner of the group
-    /// as well as an admin and member of the group.
+    /// Default `GroupCreateOpts` for common use cases.
+    ///
+    /// The user who calls [group_create](trait.GroupOps.html#tymethod.group_create) will be the owner of the group
```
[the remark this answers is not in the document]
Comment (giarc3) [reply src/group.rs]: Good catch, added

File: src/group.rs [hunk 3]
```diff
@@ -11,55 +13,59 @@ use crate::{
...
+    /// List of users to add as admins of the group. Even if `add_as_admin` is false, the calling user will be added as an admin if they are in this list.
     admins: Vec<UserId>,
-    // list of users to add as members of the group.
-    // note: even if `add_as_member` is false, the calling user will be added as a member if they are in this list.
+    /// List of users to add as members of the group. Even if `add_as_member` is false, the calling user will be added as a member if they are in this list.
     members: Vec<UserId>,
-    // true - group's private key will be marked for rotation
-    // false (default) - group's private key will not be marked for rotation
+    /// - `true` - group's private key will be marked for rotation
+    /// `false` (default) - group's private key will not be marked for rotation
```
[the remark this answers is not in the document]
Comment (giarc3) [reply src/group.rs]: Good catch, fixed

File: src/group.rs [hunk 4]
```diff
@@ -141,143 +149,287 @@ impl GroupCreateOpts {
...
-    /// Remove a list of users as admins from the group.
+    /// Removes administrators from a group.
+    ///
+    /// Returns lists of successful and failed removals.
     ///
     /// # Arguments
-    /// - `id` - ID of the group
-    /// - `revoke_list` - List of user IDs to remove as admins
+    /// - `id` - ID of the group to remove administrators from
+    /// - `revoke_list` - List of users to remove as group administrators
```
Comment (clintfred) [inline src/group.rs:382]: I would almost call this `to_remove` or something. Perhaps this outside the scope of the doc rework though.

File: src/lib.rs [hunk 5]
```diff
@@ -17,14 +17,37 @@
 //!
 //! # Group Operations
 //!
-//! Groups are one of the many differentiating features of the IronCore platform. This SDK allows for easy management of your cryptographic
-//! groups. Groups can be created, updated, and deleted along with management of a group's administrators and members.
+//! Groups are one of the many differentiating features of the IronCore platform. Groups are collections of users who share access permissions.
```
Comment (clintfred) [inline src/lib.rs:20]: Here's some additional text that may be useful here. This comes from the DataControl Guide.

> Groups are collections of users. A user in the context of a group is called a member. Groups can have any number of members. Group membership can be modified by users designated as group administrators. Adding a user to a group is a series of cryptographic operations involving the administrating user’s keys, the group’s keys, and the new member’s public key.

Comment (giarc3) [reply src/lib.rs:20]: I added some of those parts to this section. I could use feedback on the wording and content of this section if you have any.

File: src/group.rs [hunk 6]
```diff
@@ -141,143 +149,287 @@ impl GroupCreateOpts {
...
+    /// let rotate_result = sdk.group_rotate_private_key(&group_id).await?;
+    /// let new_rotation = rotate_result.needs_rotation();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn group_rotate_private_key(&self, id: &GroupId) -> Result<GroupUpdatePrivateKeyResult>;
+
+    /// Adds members to a group.
+    ///
+    /// Returns lists of successful and failed additions.
```
[the remark this answers is not in the document]
Comment (clintfred) [reply src/group.rs:283]: I think what I mostly don't like is "lists of". But /shrug

[the remark this answers is not in the document]
Comment (giarc3) [reply src/group.rs]: Oh, sure. Changed

File: src/lib.rs (hunk 5 above)
Comment (clintfred) [reply src/lib.rs:20]:
```suggestion
//! Groups are one of the many differentiating features of the DataControl platform. Groups are collections of users who share access permissions.
```

**Pull request merged**